azure-kusto-data/source/client.ts

```diff
@@ -142,6 +143,11 @@ export class KustoClient {
             throw error;
         }
 
+        // TODO: temp - axios shoud have thrown error in this case, need to understand why it didn't for some customers
+        if(axiosResponse.status != 200){
```

Comment:
most HTTP rest libraries don't throw on non-200 statuses, thats up to consumers to implement

https://github.com/axios/axios/issues/41

Reply:
if axios throws it generally means it was a client side error (client side timeout etc)

Reply:
There is a built in validateStatus of axios, I now set it explicitly (even though for me it works without that)